### PR TITLE
fix(webpack): `MultiCompiler` extends `Tapable`

### PR DESCRIPTION
**Pull request by alan-agius4**
https://github.com/webpack/webpack/blob/69e0844028950e1316fb4ed9f4ac4593ecc205b3/lib/MultiCompiler.js#L12

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/webpack/webpack/blob/69e0844028950e1316fb4ed9f4ac4593ecc205b3/lib/MultiCompiler.js#L12>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by bumbleblym** [comment]
@andy-ms build is failing due to [new tslint (5.8.0) rule: no-redundant-jsdoc](https://github.com/palantir/tslint/blob/e78f9c233935ea071fbc6bec1f67336eac1f9e41/CHANGELOG.md#tada-features)

File: types/webpack/index.d.ts
Comment (mohsen1) [inline types/webpack/index.d.ts]: can you review changes and make sure we're adding all new APIs?

Comment (alan-agius4) [reply types/webpack/index.d.ts]: I updated some stuff that I found missing!

**Comment by mohsen1** [comment]
This has become a large change. 

@tkqubo @bumbleblym @bcherny @tommytroylin @mohsen1 @jcreamer898 @ahmed-taj

Please review

**Comment by mohsen1** [comment]
+1

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Review by z0al — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks everyone!

File: types/webpack/index.d.ts
Comment (thasmo) [inline types/webpack/index.d.ts]: Shouldn't this be `boolean` instead of `true`?

**Comment by alan-agius4** [comment]
Good spot

File: types/webpack/index.d.ts
Comment (thasmo) [inline types/webpack/index.d.ts]: Should be `boolean`, I guess.

**Comment by alan-agius4** [comment]
#21342

File: types/webpack/index.d.ts
Comment (alan-agius4) [reply types/webpack/index.d.ts]: FIxed in #21342

File: types/webpack/index.d.ts
Comment (alan-agius4) [reply types/webpack/index.d.ts]: FIxed in #21342

**Pull request merged**
